src/owner.rs

```diff
@@ -0,0 +1,357 @@
+            Owner::User(try!(User::find_by_login(conn, name).map_err(|_|
+                human(format!("could not find user with login `{}`", name))
+            )))
+        };
+        Ok(owner)
+    }
+
+    /// Find the owner by name, with the intent of adding it as an owner.
+    ///
+    /// This differs from find_by_name in that in the case of a Team,
+    /// it will verify the req_user is on the team first.
+    ///
+    /// If the req_user is on the Team, it will create the team in the DB
+    /// if it is not already present. When this occurs, this will set the
+    /// One True Name of the team. All future references to the team must
+    /// use the exact casing provided here. If a different casing is provided
+    /// to this method, we may still succeed if Github returns us the same ID
+    /// as the One True Name. However in this case, the One True Name will
+    /// still be selected.
+    pub fn find_by_name_for_add(conn: &Connection, name: &str, req_user: &User)
+        -> CargoResult<Owner> {
+        if !name.contains(":") {
+            return Ok(Owner::User(try!(User::find_by_login(conn, name).map_err(|_|
+                human(format!("could not find user with login `{}`", name))
+            ))));
+        }
+
+        // We're working with a Team, try to just get it out of the DB.
+        if let Ok(team) = Team::find_by_name(conn, name) {
+            return if try!(team.contains_user(req_user)) {
+                Ok(Owner::Team(team))
+            } else {
+                Err(human(format!("only members of {} can add it as an owner", name)))
+            };
+        }
+
+        // Failed to retrieve from the DB, must be a new Team, try to add it.
+        Ok(Owner::Team(try!(Team::create(conn, name, req_user))))
+    }
+
+    pub fn kind(&self) -> i32 {
+        match *self {
+            Owner::User(_) => 0,
+            Owner::Team(_) => 1,
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        match *self {
+            Owner::User(ref user) => &user.gh_login,
+            Owner::Team(ref team) => &team.name,
+        }
+    }
+
+    pub fn id(&self) -> i32 {
+        match *self {
+            Owner::User(ref user) => user.id,
+            Owner::Team(ref team) => team.cargo_id,
+        }
+    }
+
+    pub fn encodable(self) -> EncodableOwner {
+        match self {
+            Owner::User(User { id, email, name, gh_login, avatar, .. }) => {
+                EncodableOwner {
+                    id: id,
+                    login: gh_login,
+                    email: email,
+                    avatar: avatar,
+                    name: name,
+                    kind: String::from("user"),
+                }
+            }
+            Owner::Team(Team { cargo_id, name, .. }) => {
+                EncodableOwner {
+                    id: cargo_id,
+                    login: name,
+                    email: None,
+                    avatar: None,
+                    name: None,
```

Comment:
ok I have this back in cache now. `name` is the pretty-printed-name. e.g. "Alexis Beingessner". So it doesn't super make sense to produce that here? (it will make cargo produce a silly output when you list owners).

We do want a duplicate of `login`, it's unfortunate that `name` is taken!